src/lib/storage/up-storage.js

```diff
@@ -227,10 +227,67 @@ class ProxyStorage {
     headers[acceptEncoding] = headers[acceptEncoding] || 'gzip';
     // registry.npmjs.org will only return search result if user-agent include string 'npm'
     headers[userAgent] = headers[userAgent] || `npm (${this.userAgent})`;
+
+    return this._setAuth(headers);
+  }
+
+  /**
+   * Validate configuration auth and assign Header authorization
+   * @param {Object} headers
+   * @return {Object}
+   * @private
+   */
+  _setAuth(headers) {
+
+    if (_.isUndefined(this.config.auth)) {
+      return headers;
+    }
+
+    // if header Authorization assigns this has precedence
+    if (headers['authorization']) {
```

Comment:
I think you could combine this return with the one above. At the end, they return the same.